### PR TITLE
allow specific image tags

### LINKED ISSUE #62: [cetic/zabbix] specific image tags for artifacts
**Is your feature request related to a problem? Please describe.**

Sometimes it is necessary to use specific versions of zabbix docker images if there are bugs in recent versions.
A good example: https://support.zabbix.com/browse/ZBXNEXT-7778 (downgrading to 6.0.4 helps to solve the problem)

**Describe the solution you'd like**

Allow the definition of specific versions.

### PR DESCRIPTION
**Pull request by scoopex**
introduce specific image tags

### CODE REVIEW

**Comment by aeciopires** [comment]
Hi, @scoopex! 

I understand your necessity. The PR is cool for me. Do you agree @sa-ChristianAnton?

**Comment by aeciopires** [comment]
By default specific tags is not defined, but with this PR is possible to use.

**Comment by scoopex** [comment]
> Hi @scoopex!
> 
> Sorry. I reviewed and approved quickly. Talking with @sa-ChristianAnton I saw that the parameters you are setting as follows.
> 
> ```yaml
>           image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver_image_tag }}"
>           image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent_image_tag }}"
>           image: "{{ .Values.zabbixweb.image.repository }}:{{ .Values.zabbixweb_image_tag }}"
>           image: "{{ .Values.zabbixwebservice.image.repository }}:{{ .Values.zabbixwebservice_image_tag }}"
> ```
> 
> We think it would be more interesting to define it as follows.
> 
> ```yaml
>           image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver.image.tag }}"
>           image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent.image.tag }}"
>           image: "{{ .Values.zabbixweb.image.repository }}:{{ .Values.zabbixweb.image.tag }}"
>           image: "{{ .Values.zabbixwebservice.image.repository }}:{{ .Values.zabbixwebservice.image.tag }}"
> ```
> 
> The same idea applies to snippets with the if statement. Example:
> 
> Before:
> 
> ```yaml
> {{- if .Values.zabbixagent_image_tag }}
> ```
> 
> After:
> 
> ```yaml
> {{- if .Values.zabbixagent.image.tag }}
> ```
> 
> Another question: could you put these specific parameters in the values.yaml file (even if commented) with the same value as the tag? This makes it more intuitive for other people to use.

Yes of course, i changed that.

File: templates/deployment-zabbix-server.yaml [hunk 1]
```diff
@@ -49,7 +49,11 @@ spec:
       {{- if gt (.Values.zabbixserver.replicaCount | int) 1 }}
       initContainers:
         - name: init-wait-for-database-schema
+          {{- if .Values.zabbixserver_image.tag }}
```
Comment (aeciopires) [inline templates/deployment-zabbix-server.yaml:52]: Hi @scoopex!

Can you change `{{- if .Values.zabbixserver_image.tag }}` by `{{- if .Values.zabbixserver.image.tag }}`?

File: templates/deployment-zabbix-server.yaml [hunk 2]
```diff
@@ -49,7 +49,11 @@ spec:
...
+          image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver_image.tag }}"
```
Comment (aeciopires) [inline templates/deployment-zabbix-server.yaml:53]: Hi @scoopex!

Can you change `image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver_image.tag }}"` by `image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver.image.tag }}"`?

File: templates/deployment-zabbix-server.yaml [hunk 3]
```diff
@@ -63,7 +67,11 @@ spec:
         - name: zabbix-server
           resources:
 {{- toYaml .Values.zabbixserver.resources | nindent 12 }}
+          {{- if .Values.zabbixserver_image.tag }}
```
Comment (aeciopires) [inline templates/deployment-zabbix-server.yaml:70]: Hi @scoopex!

Can you change

{{- if .Values.zabbixserver_image.tag }}

by

{{- if .Values.zabbixserver.image.tag }}

?

File: templates/deployment-zabbix-server.yaml [hunk 4]
```diff
@@ -63,7 +67,11 @@ spec:
...
+          image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver_image.tag }}"
```
Comment (aeciopires) [inline templates/deployment-zabbix-server.yaml:71]: Hi @scoopex!

Can you change

`image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver_image.tag }}"`

by

`image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver.image.tag }}"`

?

File: templates/deployment-zabbix-server.yaml [hunk 5]
```diff
@@ -109,7 +117,11 @@ spec:
 {{- toYaml .Values.zabbixagent.resources | nindent 12 }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- if .Values.zabbixagent_image.tag }}
```
Comment (aeciopires) [inline templates/deployment-zabbix-server.yaml:120]: Hi @scoopex!

Can you change

{{- if .Values.zabbixagent_image.tag }}

by

{{- if .Values.zabbixagent.image.tag }}

?

File: templates/deployment-zabbix-server.yaml [hunk 6]
```diff
@@ -109,7 +117,11 @@ spec:
...
+          image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent_image.tag }}"
```
Comment (aeciopires) [inline templates/deployment-zabbix-server.yaml:121]: Hi @scoopex!

Can you change

`image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent_image.tag }}"`

by

`image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent.image.tag }}"`

?

File: templates/deployment-zabbix-web.yaml [hunk 7]
```diff
@@ -51,7 +51,11 @@ spec:
           {{- toYaml .Values.zabbixweb.resources | nindent 10 }}
         securityContext:
           {{- toYaml .Values.securityContext | nindent 12 }}
+        {{- if .Values.zabbixweb_image.tag }}
```
Comment (aeciopires) [inline templates/deployment-zabbix-web.yaml:54]: Hi @scoopex!

Can you change

{{- if .Values.zabbixweb_image.tag }}

by

{{- if .Values.zabbixweb.image.tag }}

?

File: templates/deployment-zabbix-web.yaml [hunk 8]
```diff
@@ -51,7 +51,11 @@ spec:
...
+        image: "{{ .Values.zabbixweb.image.repository }}:{{ .Values.zabbixweb_image.tag }}"
```
Comment (aeciopires) [inline templates/deployment-zabbix-web.yaml:55]: Hi @scoopex!

Can you change

`image: "{{ .Values.zabbixweb.image.repository }}:{{ .Values.zabbixweb_image.tag }}"`

by

`image: "{{ .Values.zabbixweb.image.repository }}:{{ .Values.zabbixweb.image.tag }}"`

?

File: templates/deployment-zabbix-webservice.yaml [hunk 9]
```diff
@@ -51,7 +51,11 @@ spec:
           {{- toYaml .Values.zabbixwebservice.resources | nindent 10 }}
         securityContext:
           {{- toYaml .Values.securityContext | nindent 12 }}
+        {{- if .Values.zabbixwebservice_image.tag }}
```
Comment (aeciopires) [inline templates/deployment-zabbix-webservice.yaml:54]: Hi @scoopex!

Can you change

`{{- if .Values.zabbixwebservice_image.tag }}`

by

`{{- if .Values.zabbixwebservice.image.tag }}`

?

File: templates/deployment-zabbix-webservice.yaml [hunk 10]
```diff
@@ -51,7 +51,11 @@ spec:
...
+        image: "{{ .Values.zabbixwebservice.image.repository }}:{{ .Values.zabbixwebservice_image.tag }}"
```
Comment (aeciopires) [inline templates/deployment-zabbix-webservice.yaml:55]: Hi @scoopex!

Can you change

`image: "{{ .Values.zabbixwebservice.image.repository }}:{{ .Values.zabbixwebservice_image.tag }}"`

by

`image: "{{ .Values.zabbixwebservice.image.repository }}:{{ .Values.zabbixwebservice.image.tag }}"`

?

File: templates/job-init-db-schema.yaml [hunk 11]
```diff
@@ -14,7 +14,11 @@ spec:
     spec:
       containers:
       - name: init-db-schema
-        image: {{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbix_image_tag }}
+        {{- if .Values.zabbixserver_image.tag }}
```
Comment (aeciopires) [inline templates/job-init-db-schema.yaml:17]: Hi @scoopex!

Can you change

`{{- if .Values.zabbixserver_image.tag }}`

by

`{{- if .Values.zabbixserver.image.tag }}`

?

File: templates/job-init-db-schema.yaml [hunk 12]
```diff
@@ -14,7 +14,11 @@ spec:
...
+        image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver_image.tag }}"
```
Comment (aeciopires) [inline templates/job-init-db-schema.yaml:18]: Hi @scoopex!

Can you change

`image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver_image.tag }}"`

by

`image: "{{ .Values.zabbixserver.image.repository }}:{{ .Values.zabbixserver.image.tag }}"`

?

File: templates/statefulset-zabbix-proxy.yaml [hunk 13]
```diff
@@ -87,7 +91,12 @@ spec:
 {{- toYaml .Values.zabbixproxy.resources | nindent 12 }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- if .Values.zabbixproxy_image.tag }}
+          image: "{{ .Values.zabbixproxy.image.repository }}:{{ .Values.zabbixproxy_image.tag }}"
```
Comment (aeciopires) [inline templates/statefulset-zabbix-proxy.yaml:95]: Hi @scoopex!

Can you change

`image: "{{ .Values.zabbixproxy.image.repository }}:{{ .Values.zabbixproxy_image.tag }}"`

by

`image: "{{ .Values.zabbixproxy.image.repository }}:{{ .Values.zabbixproxy.image.tag }}"`

?

File: templates/statefulset-zabbix-proxy.yaml [hunk 14]
```diff
@@ -87,7 +91,12 @@ spec:
...
+          {{- else }}
           image: "{{ .Values.zabbixproxy.image.repository }}:{{ .Values.zabbix_image_tag }}"
+          {{- end }}
+
```
Comment (aeciopires) [inline templates/statefulset-zabbix-proxy.yaml:99]: Hi @scoopex!

Please, can you to remove  this line?

File: templates/statefulset-zabbix-proxy.yaml [hunk 15]
```diff
@@ -87,7 +91,12 @@ spec:
 {{- toYaml .Values.zabbixproxy.resources | nindent 12 }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- if .Values.zabbixproxy_image.tag }}
```
Comment (aeciopires) [inline templates/statefulset-zabbix-proxy.yaml:94]: Hi @scoopex!

Can you change

`{{- if .Values.zabbixproxy_image.tag }}`

by

`{{- if .Values.zabbixproxy.image.tag }}`

?

File: templates/statefulset-zabbix-proxy.yaml [hunk 16]
```diff
@@ -44,7 +44,11 @@ spec:
 {{- toYaml .Values.zabbixagent.resources | nindent 12 }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- if .Values.zabbixagent_image.tag }}
+          image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent_image.tag }}"
```
Comment (aeciopires) [inline templates/statefulset-zabbix-proxy.yaml:48]: Hi @scoopex!

Can you change

`image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent_image.tag }}"`

by

`image: "{{ .Values.zabbixagent.image.repository }}:{{ .Values.zabbixagent.image.tag }}"`

?

File: templates/statefulset-zabbix-proxy.yaml [hunk 17]
```diff
@@ -44,7 +44,11 @@ spec:
 {{- toYaml .Values.zabbixagent.resources | nindent 12 }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
+          {{- if .Values.zabbixagent_image.tag }}
```
Comment (aeciopires) [inline templates/statefulset-zabbix-proxy.yaml:47]: Hi @scoopex!

Can you change

`{{- if .Values.zabbixagent_image.tag }}`

by

`{{- if .Values.zabbixagent.image.tag }}`

?

**Review by aeciopires — CHANGES_REQUESTED** [review]
Hi @scoopex!

I did some comments.

Please, could you put these specific parameters in the values.yaml file (even if commented) with the same value as the tag `zabbix_image_tag`? This makes it more intuitive for other people to use.

File: templates/deployment-zabbix-server.yaml (hunk 1 above)
Comment (scoopex) [reply templates/deployment-zabbix-server.yaml]: Sure, done :-)

File: templates/deployment-zabbix-server.yaml (hunk 2 above)
Comment (scoopex) [reply templates/deployment-zabbix-server.yaml]: Sure, done :-)

File: templates/deployment-zabbix-server.yaml (hunk 3 above)
Comment (scoopex) [reply templates/deployment-zabbix-server.yaml]: Sure, done :-)

File: templates/deployment-zabbix-server.yaml (hunk 4 above)
Comment (scoopex) [reply templates/deployment-zabbix-server.yaml]: Sure, done :-)

File: templates/deployment-zabbix-server.yaml (hunk 5 above)
Comment (scoopex) [reply templates/deployment-zabbix-server.yaml]: Sure, done :-)

File: templates/deployment-zabbix-server.yaml (hunk 6 above)
Comment (scoopex) [reply templates/deployment-zabbix-server.yaml]: Sure, done :-)

File: templates/deployment-zabbix-web.yaml (hunk 7 above)
Comment (scoopex) [reply templates/deployment-zabbix-web.yaml]: Sure, done :-)

File: templates/deployment-zabbix-web.yaml (hunk 8 above)
Comment (scoopex) [reply templates/deployment-zabbix-web.yaml]: Sure, done :-)

File: templates/deployment-zabbix-webservice.yaml (hunk 9 above)
Comment (scoopex) [reply templates/deployment-zabbix-webservice.yaml]: Sure, done :-)

File: templates/deployment-zabbix-webservice.yaml (hunk 10 above)
Comment (scoopex) [reply templates/deployment-zabbix-webservice.yaml]: Sure, done :-)

File: templates/job-init-db-schema.yaml (hunk 11 above)
Comment (scoopex) [reply templates/job-init-db-schema.yaml]: Sure, done :-)

File: templates/job-init-db-schema.yaml (hunk 12 above)
Comment (scoopex) [reply templates/job-init-db-schema.yaml]: Sure, done :-)

File: templates/statefulset-zabbix-proxy.yaml (hunk 13 above)
Comment (scoopex) [reply templates/statefulset-zabbix-proxy.yaml]: Sure, done :-)

File: templates/statefulset-zabbix-proxy.yaml (hunk 15 above)
Comment (scoopex) [reply templates/statefulset-zabbix-proxy.yaml]: Sure, done :-)

File: templates/statefulset-zabbix-proxy.yaml (hunk 16 above)
Comment (scoopex) [reply templates/statefulset-zabbix-proxy.yaml]: Sure, done :-)

File: templates/statefulset-zabbix-proxy.yaml (hunk 17 above)
Comment (scoopex) [reply templates/statefulset-zabbix-proxy.yaml]: Sure, done :-)

**Review by aeciopires — CHANGES_REQUESTED** [review]
Hi @scoopex!

Thanks for apply changes suggested.
Good idea define specific tags with `null` value by default. 

For finish, can you run command `make gen-docs` to update `README.md` file with new parameters?

After this, I will publish the new version of helm chart.

**Comment by scoopex** [comment]
Pushed "make gen-docs" update.

**Review by aeciopires — APPROVED** [review]
Hi @scoopex!

Thanks for your contribution.

**Pull request merged**